ring/sampler_gaussian.go

```diff
@@ -137,15 +182,22 @@ func randFloat64(randomBytes []byte) float64 {
 // to use a secure PRNG instead of math/rand.
 func (g *GaussianSampler) normFloat64() (float64, uint64) {
 
+	ptr := g.ptr
+	buff := g.randomBufferN
+	prng := g.prng
+	buffLen := uint64(len(buff))
+
 	for {
 
-		if g.ptr == uint64(len(g.randomBufferN)) {
-			g.prng.Read(g.randomBufferN)
-			g.ptr = 0
+		if ptr == buffLen {
+			if _, err := prng.Read(buff); err != nil {
+				panic(err)
+			}
+			ptr = 0
 		}
 
-		juint32 := binary.BigEndian.Uint32(g.randomBufferN[g.ptr : g.ptr+4])
-		g.ptr += 8
+		juint32 := binary.LittleEndian.Uint32(buff[ptr : ptr+4])
+		ptr += 8
```

Comment:
is it normal the ptr is getting incremented by 8 here instead of 4 ?

Reply:
Yes, to avoid a misalignment of the buffer pointer, else it could overflow N